webdriver/tests/support/fixtures.py

```diff
@@ -38,6 +41,20 @@ def pytest_generate_tests(metafunc):
         if marker:
             metafunc.parametrize("capabilities", marker.args, ids=None)
 
+    if "bidi" in metafunc.fixturenames:
+        marker = metafunc.definition.get_closest_marker(name="bidi")
+        if marker:
+            metafunc.parametrize("bidi", marker.args, ids=None)
+
+# Ensure that the event loop is restarted once per session rather than the default  of once per test
+# if we don't do this, tests will try to reuse a closed event loop and fail with an error that the "future
+# belongs to a different loop"
+@pytest.fixture(scope="session")
+def event_loop():
+    """Change event_loop fixture to session level."""
+    loop = asyncio.get_event_loop_policy().new_event_loop()
+    yield loop
+    loop.close()
```

Comment:
With the dedicated session approach this can also become part of the`bidi_session` fixture.

Reply:
I created the bidi_session fixture in the same fixtures.py. Not sure how to make this "become part of the ```bidi_session``` fixture".

Reply:
Lets move this to the new issue/PR. Mind moving over all the remaining open conversations?

Reply:
Will do.